src/libraries/icubmod/embObjMotionControl/eomcParser.h

```diff
@@ -56,6 +57,226 @@ namespace yarp {
 //    PidAlgo_currentInnerLoop =2
 //} PidAlgorithmType_t;
 
+typedef enum
+{
+    motor_temperature_sensor_pt100       = 0,
+    motor_temperature_sensor_pt1000      = 1,
+    motor_temperature_sensor_none        = 255
+} motor_temperatureSensorTypeEnum_t;
+
+class ITemperatureSensor
+{
+
+public:
+
+    virtual double convertTempCelsiusToRaw(const double temperature) = 0;
+
+    virtual double convertRawToTempCelsius(const double temperature) = 0;
+
+    virtual motor_temperatureSensorTypeEnum_t getType() = 0;
+};
+
+class TemperatureSensorPT100 : public ITemperatureSensor
+{
+private:
+    // resistors of the voltage divider bridge
+    int _r_1;
+    int _r_2;
+    int _r_3;
+    
+
+    double _ptc_offset;    // offset of the temperature sensor line
+    double _ptc_gradient;  // slope/gradient of the temperature sensor line
+    double _pga_gain;      // ADC gain set for the tdb (temperature detection board)
+
+    int _vcc;               // vcc that enters the voltage divider bridge
+    double _resolution_pga; // resolution of the internal pga of the tdb
+    double _resolution_tdb; // resolution used for the raw value for the output of the tdb
+
+    double _half_bridge_resistor_coeff;
+
+    bool isConfigured = false;
+
+public:
+
+    TemperatureSensorPT100()
+    {
+        _r_1 = 4700;
+        _r_2 = 4700;
+        _r_3 = 100; 
+
+        _ptc_offset = 100.0;
+        _ptc_gradient = 0.3851;
+        _pga_gain = 2;
+        _vcc = 5;
+        _resolution_pga = 2.048;
+        _resolution_tdb = 32767; 
+
+        _half_bridge_resistor_coeff = (double)_r_3 / (double)(_r_2 + _r_3);
+        isConfigured = true;
```

Comment:
`isconfigured` is no longer useful. Its value is always true.
Please remove it.

Reply:
Check last commit. Update as requested

Reply:
ok